lib/Haskell/Prim/Absurd.agda

```diff
@@ -0,0 +1,34 @@
+
+module Haskell.Prim.Absurd where
+
+open import Agda.Builtin.Nat
+open import Agda.Builtin.List
+open import Agda.Builtin.Unit
+open import Agda.Builtin.Reflection renaming (bindTC to _>>=_; absurd to absurdP)
+open import Agda.Builtin.Equality
+
+private
+
+  pattern vArg x = arg (arg-info visible relevant) x
+
+  variable
+    a b : Set
+
+  _$_ : (a → b) → a → b
+  f $ x = f x
+
+  refute : Nat → Term
+  refute i = def (quote _$_) (vArg (pat-lam (absurd-clause (vArg absurdP ∷ []) ∷ []) []) ∷ vArg (var i []) ∷ [])
```

Comment:
No, it's important that it's an absurd lambda and not `⊥-elim`, since the absurd lambda works for *any* empty type, not just `⊥`. The reason you need `_$_` is to rule out a dependent function type for the absurd lambda.

Reply:
...and you never *need* it to be a dependent function since we apply it to a variable in the context. An example:
```agda
bla : (i : Empty) -> T i
bla i = (λ ()) i
```
Here, you might think the absurd lambda should have the dependent type `(j : Empty) -> T j`, but since `i` is already in the context, the non-dependent `Empty -> T i` works just as well. Without restricting to the non-dependent type the type checker doesn't know which type to choose and leaves unsolved constraints.